libvirt/domain_def.go

```diff
@@ -17,7 +17,7 @@ func getXMLDomainDefFromLibvirt(domain *libvirt.Domain) (libvirtxml.Domain, erro
 		return libvirtxml.Domain{}, fmt.Errorf("Error retrieving libvirt domain XML description: %s", err)
 	}
 
-	domainDef := newDomainDef()
+	domainDef := newDomainDef("x86_64")
```

Comment:
I was thinking that looks a bit hardcoded the `x86_64` here.

Personally I would prefer having the `newDomainDef()` with void parameter.

After when the domainDef is created modify the code depending the arch.

I dunno if you did this initially.

What do you think @dmacvicar ?

Reply:
I think the best way to get a good default is to use ` virConnectGetDomainCapabilities`, which gives you the arch for the domains before creating it.

Originally I thought that we get the capabilities after we create the empty definition (and we are, but that is complicated to explain). However, in the "Read" section, we call:

```golang
domainDef, err := newDomainDefForConnection(virConn, d)
```

Definition options are overwritten with the capabilities one if they are not specified by the domain definition. I really wonder about this part. Perhaps has to do that "Read" is called at the end of "Create", and this makes sure computed attributes are set if not specified in the terraform data. Dunno.

//cc @Prashanth684 

This method is a method that returns a definition having access to the connection and capabilities, which means it has access to the default architecture.

Perhaps using this call in the "Create" section instead of the vanilla `newDomainDef` is a good solution. We can have `newDomainDefForConnection` set the default bootloader just after it sets the arch (either specified, or from the capabilities).

I don't know if using `newDomainDefForConnection` in "Create" has other consequences.

Reply:
@MalloZup  @dmacvicar makes sense...i did put it in `newDomainDefForConnection` first but thought to put it in `newDomainDef` later in case we miss some places where we set it, but doesn't look like it. i will make that change. i didn't very much like hardcoding it as well.

Yes , that `Read` call seems to just fill the schema with all the computed attributes , but i also wonder if we need to set things like `OS.Type.Machine` in `newDomainDefForConnection` since libvirt should figure those automatically, but that warrants a separate discussion.